### ISSUE TITLE
is:npm hidden command is broken due to invalid config arguments (duplicate config)

### ISSUE THREAD

**Issue by hvisser** [firebase/firebase-tools#6132]
I'm using the (hidden) is:npm function to check the latest available version of firebase-tools in my Github action.
I understand that this might be a bit on the edge of supported, however the `is:npm` function in `firepit` seems broken at the moment, which is probably not intended :)

### [REQUIRED] Environment info

**firebase-tools:**

firebase-tools standalone 12.4.4

**Platform:**

ubuntu, did not test on other platforms, but could be equally affected.

### [REQUIRED] Test case

run `firebase is:npm`

### [REQUIRED] Steps to reproduce

run `firebase is:npm`

### [REQUIRED] Expected behavior

Calls the embedded npm

### [REQUIRED] Actual behavior

Returns an error similar to 
```
Error: double-loading config "/home/botteaap/.cache/firebase/runtime/npmrc" as "global", previously loaded as "user"
    at Config.#loadObject (/snapshot/firepit/node_modules/npm/node_modules/@npmcli/config/lib/index.js:571:13)
    at /snapshot/firepit/node_modules/npm/node_modules/@npmcli/config/lib/index.js:617:31
    at async Config.#loadFile (/snapshot/firepit/node_modules/npm/node_modules/@npmcli/config/lib/index.js:616:5)
    at async Config.load (/snapshot/firepit/node_modules/npm/node_modules/@npmcli/config/lib/index.js:287:5)
    at async Npm.#load (/snapshot/firepit/node_modules/npm/lib/npm.js:195:5)
    at async /snapshot/firepit/node_modules/npm/lib/cli-entry.js:40:5
```
The root cause seems to be that the same `npmrc` is passed to `npm` here https://github.com/firebase/firebase-tools/blob/master/standalone/firepit.js#L186 which apparently is no longer allowed.

**Comment by hvisser** [comment]
Actually, this seems a bit more severe as I'm also seeing the same error when running `firebase deploy` with the standalone version (at the point where it's calling out to run `npm run lint`)

**Comment by hvisser** [comment]
I've downgraded to 12.4.3 which doesn't have this bug. I also reproduced this on mac os in the mean time. My function deploys were failing with 12.4.4 because of this too.

**Comment by hvisser** [comment]
Any updates on this maybe? I'm stuck on 12.4.3 because my functions do not deploy with the the firepit version.

**Comment by cookii-king** [comment]
😕 I'm experiencing the same issue as well. I was really looking forward to testing out the new v2, but unfortunately, I'm encountering difficulties deploying the function. 

![Screenshot 2023-08-18 at 15:49:38](https://github.com/firebase/firebase-tools/assets/83700098/fa67fe47-a179-4397-9dbf-09904af4e81c)

**Comment by Jabi7** [comment]
Has this been resolved for anyone? i'm stuck on this error for days. please help... :'(

`i  deploying functions
Running command: npm --prefix ./functions/ run lint
Error: double-loading config "/home/crow/.cache/firebase/runtime/npmrc" as "global", previously loaded as "user"
at Config.#loadObject (/snapshot/firepit/node_modules/npm/node_modules/@npmcli/config/lib/index.js:571:13)
at /snapshot/firepit/node_modules/npm/node_modules/@npmcli/config/lib/index.js:617:31
at async Config.#loadFile (/snapshot/firepit/node_modules/npm/node_modules/@npmcli/config/lib/index.js:616:5)
at async Config.load (/snapshot/firepit/node_modules/npm/node_modules/@npmcli/config/lib/index.js:287:5)
at async Npm.#load (/snapshot/firepit/node_modules/npm/lib/npm.js:195:5)
at async /snapshot/firepit/node_modules/npm/lib/cli-entry.js:40:5

Error: functions predeploy error: Command terminated with non-zero exit code 1`

**Comment by hvisser** [comment]
I was attempting a PR to fix this, but it seems like bumping the `npm` dependency is the root cause. Even if I remove the duplicate config line that I linked to, I'm not getting any output running `firebase is:npm help`. When I downgrade the `npm` dependency things start working again. @joehan I hope you have any ideas and that this info is useful to you?

@cookii-king The only workaround so far, if you want to use the standalone version of the firebase cli, is to stick to version 12.4.3 https://firebase.tools/ has instructions on how to get a specific version. The version installed through npm doesn't have this issue (I think, I hope 😅 )

**Comment by hvisser** [comment]
Digging a bit deeper, there are errors when running the `npm pkg` step. Those in turn are related to https://github.com/vercel/pkg/issues/1291. The newer `npm` dependency uses esm modules. What seems to be happening is that not all dependencies used for the npm cli are compiled correctly, hence the broken `npm`. The built-in node version using `is:node` also seems broken by the way. 

I guess the short term solution would be to revert the npm dependency bump which would unbreak standalone versions of `firebase-tools` after `12.4.3`.

**Comment by joehan** [comment]
@hvisser Thanks for all the research into this, I'm looking into a fix now (hopefully without going all the way back to npm 6.x). Apologies for the delay on this - I had deprioritized it in my head since `is:npm` isn't an officially supported command, but since this seems to affect _any_ call to npm from the standalone version of the CLI, it is much more concerning.

**Comment by hvisser** [comment]
Thanks @joehan I didn't realise that either when I filed the issue initially, so I completely understand.

**Comment by Masterxilo** [comment]
I had the same problem today. Installing via "curl -sL https://firebase.tools | bash" we apparently still get a bad version...

**Comment by joehan** [comment]
@Masterxilo There has not been a new version released since we fixed this bug, so that is expected. This should be resolved in the next version

**Comment by hvisser** [comment]
@joehan Turns out this is not completely fixed, see PR #6367. LMK if you need a separate issue for that too.

**Comment by gkhnyrdm** [comment]
anyone fix this issue?

**Comment by tdelam** [comment]
Receiving this issue as well

`node --version && firebase --version
v20.5.0
12.5.2`

**Comment by tdelam** [comment]
> I've downgraded to 12.4.3 which doesn't have this bug. I also reproduced this on mac os in the mean time. My function deploys were failing with 12.4.4 because of this too.

This worked for me too.

**Comment by eszabo12** [comment]
node --version && firebase --version

v21.5.0
12.4.7

**Comment by joehan** [comment]
Hey @eszabo12 - this issue was fixed in 12.6.0. Please update to a more recent version of firebase-tools.

**Comment by joehan** [comment]
@eszabo12 Since that's a different error, could you open a new issue and include the `firepit-log.txt` file? It will be difficult to help otherwise.
